compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -773,9 +772,19 @@ checkDefTypeSyn DefTypeSyn{synParams,synType} = do
   where
     base = checkType synType KStar
 
+
+checkIface :: MonadGamma m => DefInterface -> m ()
+checkIface DefInterface{intName, intChoices} = do
+  checkUnique (EDuplicateInterfaceChoiceName intName) $ NM.names intChoices
+  forM_ intChoices checkIfaceChoice
+
+checkIfaceChoice :: MonadGamma m => InterfaceChoice -> m ()
+checkIfaceChoice InterfaceChoice{ifcRetType} =
+  checkType ifcRetType KStar
```

Comment:
Should we also check the kind of the argument type?

Reply:
yep, added the check.